main.go

```diff
@@ -97,7 +109,13 @@ func resolveConfig() (*config.Config, bool) {
 	flag.Parse()
 
 	if *printVersion {
-		return nil, true
+		otherOptions.PrintVersion = true
+		return nil, otherOptions
+	}
+
+	if *runOnce {
+		otherOptions.RunOnce = true
+		return nil, otherOptions
```

Comment:
Returning nil as *config.Config here causes nil pointer panic on the caller side. https://github.com/mackerelio/mackerel-agent/blob/b54cbc5/main.go#L59